src/Controller/ApplePay.php

```diff
@@ -156,4 +181,25 @@ public function process_standard_payment($order, $amount, $customer_id)
 
 	}
 
+	/**
+	 * Check User-Agent to make sure it is on Mac OS and in Safari Browser
+	 *
+	 * @return bool
+	 */
+	private function checkDeviceComptability(){
+		$user_agent = $_SERVER['HTTP_USER_AGENT'];
+
+		// Check if the Browser is Safari
+		if (stripos( $user_agent, 'Chrome') !== false) {
```

Comment:
don't you need only the safari validation?
otherwise shouldn't we add all other browsers?

Reply:
That wasn't added by me, I added it to resolve the conflict on the base branch

Reply:
All other browsers does not have the word "Safari" in user_agent string, except google chrome that have both "Safari" and "Chrome" in user_agent... so we need only to check for chrome browser in the first `if`, then all the other browsers in the `elseif`

Reply:
But why do we need assertion for Chrome? this is safari only option right?